dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/ExecutorInstrumentation.java

```diff
@@ -149,6 +150,9 @@ public boolean matches(final TypeDescription target) {
     transformers.put(
         named("execute").and(takesArgument(0, Runnable.class)),
         SetExecuteRunnableStateAdvice.class.getName());
+    transformers.put( // kotlinx.coroutines.scheduling.CoroutineScheduler
+        named("dispatch").and(takesArgument(0, Runnable.class)),
```

Comment:
I think this is what @mar-kolya is suggesting:

```suggestion
        named("dispatch").and(takesArgument(0, Runnable.class)).and(takesArgument(1, named("kotlinx.coroutines.scheduling.TaskContext"))),
```

(This will probably require reformatting.)